src/test/suite/utils/AddonDownload.test.ts

```diff
@@ -4,83 +4,218 @@ import { afterEach, describe, it } from "mocha";
 import * as fetch from "node-fetch";
 import path = require("path");
 import * as sinon from "sinon";
+import * as vscode from "vscode";
 
 import { downloadAddon } from "../../../amo/utils/addonDownload";
 import constants from "../../../config/config";
 
 describe("addonDownload.ts", async () => {
+  const workspaceFolder = path.resolve(__dirname, "..", "test_workspace");
+
   afterEach(() => {
     sinon.restore();
+    if (fs.existsSync(workspaceFolder)) {
+      fs.rmSync(workspaceFolder, { recursive: true });
+    }
   });
 
-  it("should download the xpi of the addon", async () => {
-    // mock response
-    const fakeResponse = {
-      ok: true,
-      buffer: () => {
-        return "test data";
-      },
-    };
+  const badResponse = {
+    ok: false,
+    buffer: () => {
+      return "test data";
+    },
+  };
 
-    const stub = sinon.stub();
-    stub.resolves(fakeResponse);
-    sinon.replace(fetch, "default", stub as any);
+  const goodResponse = {
+    ok: true,
+    buffer: () => {
+      return "test data";
+    },
+  };
 
-    const workspaceFolder = path.resolve(__dirname, "..", "test_workspace");
+  const addonId = "123456";
+  const addonSlug = "test-addon";
+  const downloadedFilePath = path.resolve(workspaceFolder, `${addonSlug}.xpi`);
+
+  it("should download the xpi of the addon", async () => {
     if (!fs.existsSync(workspaceFolder)) {
       fs.mkdirSync(workspaceFolder);
     }
 
-    const addonId = "123456";
-    const addonSlug = "test-addon";
-    const downloadedFilePath = path.resolve(
-      workspaceFolder,
-      `${addonSlug}.xpi`
-    );
+    const stub = sinon.stub();
+    stub.resolves(goodResponse);
+    sinon.replace(fetch, "default", stub as any);
+
+    const stub2 = sinon.stub(fs, "existsSync");
+    stub2.onFirstCall().returns(true);
 
     await downloadAddon(addonId, downloadedFilePath);
+    sinon.restore();
 
     expect(stub.calledOnce).to.be.true;
     expect(stub.calledWith(`${constants.downloadBaseURL}${addonId}`)).to.be
       .true;
 
     // wait for file to be written (there should be a better way to do this)
-    await new Promise((resolve) => setTimeout(resolve, 500));
+    await new Promise((resolve) => setTimeout(resolve, 200));
     expect(fs.existsSync(downloadedFilePath)).to.be.true;
 
     fs.rmSync(downloadedFilePath, { recursive: true });
   });
 
   it("should not make a file if the request failed", async () => {
-    const fakeResponse = {
-      ok: false,
-      buffer: () => {
-        return "test data";
-      },
-    };
+    if (!fs.existsSync(workspaceFolder)) {
+      fs.mkdirSync(workspaceFolder);
+    }
```

Comment:
Noted this in a few tests (and probably should have noticed before) - ideally we should be creating folders (or files) in a temp folder and cleaning it up again after each test.  Feel free to look into this in a follow-up issue because it's not a new thing.